include/point_one/polaris_asio_embedded_client.h

```diff
@@ -0,0 +1,657 @@
+          break;
+        case 403:
+          LOG(ERROR) << "Received an authentication forbidden (403) response "
+                        "from Polaris. Please check challenge.";
+          return false;
+        default:
+          LOG(ERROR) << "Authentication challenge response returned with status code "
+                     << status_code << ".";
+          return false;
+      }
+
+      // Process the response headers.
+      boost::asio::read_until(socket, response, "\r\n\r\n");
+      std::string header;
+      VLOG(3) << "Response:";
+      while (std::getline(response_stream, header) && header != "\r") {
+        VLOG(3) << header;
+      }
+
+      // Get the entire reply from the socket.
+      std::stringstream reply_body;
+      if (response.size() > 0) {
+        reply_body << &response;
+      }
+      boost::system::error_code error;
+      while (boost::asio::read(socket, response,
+                               boost::asio::transfer_at_least(1), error)) {
+        reply_body << &response;
+      }
+      if (error != boost::asio::error::eof) {
+        LOG(ERROR) << "Error occured while receiving auth token: "
+                   << error.message();
+        return false;
+      }
+
+      try {
+        // Parse json.
+        boost::property_tree::ptree pt;
+        boost::property_tree::read_json(reply_body, pt);
+        api_token_.access_token = pt.get<std::string>("access_token");
+        api_token_.expires_in = pt.get<double>("expires_in");
+        api_token_.issued_at = pt.get<double>("issued_at");
+        VLOG(3) << "Access token: " << api_token_.access_token;
+        VLOG(3) << "Expires in: " << std::fixed << api_token_.expires_in;
+        VLOG(3) << "Issued at: " << std::fixed << api_token_.issued_at;
+
+      } catch (std::exception &e) {
+        LOG(ERROR) << "Exception in parsing auth token response: " << e.what();
+        return false;
+      }
+
+      return true;
+    } catch (std::exception &e) {
+      LOG(ERROR) << "Exception: " << e.what();
+    }
+    return false;
+  }
+
+  // Send authentication if connection is successful, otherwise reconnect..
+  void HandleConnect(const boost::system::error_code &err) {
+    if (!err) {
+      LOG(INFO) << "Connected to Point One Navigation's Polaris Service.";
+      // The connection was successful. Send the request.
+      SendAuth();
+    } else {
+      connected_ = false;
+      connection_active_ = false;
+      LOG(ERROR) << "Failed to connect to tcp://" << connection_settings_.host
+                 << ":" << connection_settings_.port << ".";
+      ScheduleReconnect();
+    }
+  }
+
+  // Sends authentication to the tcp server.  Result is bound to async callback
+  // HandleAuthTokenWrite.
+  void SendAuth() {
+    VLOG(1) << "Authenticating with service using auth token.";
+    if (api_token_.access_token.empty()) {
+      std::string challenge  = RequestChallenge();
+      LOG(INFO) << " GOT CHALLENGE " << challenge;
```

Comment:
This seems like it should be VLOG? Or at least, a cleaner formatted message (and with no leading space) if this is something a user needs to see.